torch/distributed/utils.py

```diff
@@ -1,8 +1,92 @@
+import collections
+
+import torch
 import torch.distributed as dist
 
+
+
 def _verify_param_shape_across_processes(process_group, tensors, logger=None):
     return dist._verify_params_across_processes(process_group, tensors, logger)
 
+
+def _recursive_to(inputs, target_gpu, use_side_stream_for_tensor_copies):
```

Comment:
nit: Adding type annotations might be helpful. For example to clarify that `target_gpu` is an `int` and not a `torch.device` it seems?